app/src/main/java/com/github/emilehreich/bootcamp/MainActivity.kt

```diff
@@ -1,23 +1,55 @@
 package com.github.emilehreich.bootcamp
 
-import android.content.Intent
+
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
 import android.view.View
 import android.widget.TextView
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import java.util.concurrent.CompletableFuture
+
 
 class MainActivity : AppCompatActivity() {
+
+    companion object{
+        var database = Firebase.database
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
     }
 
-    fun greetName(view: View) {
-        val intent = Intent(this, GreetingActivity::class.java)
-        val name = findViewById<TextView>(R.id.mainText)
-        intent.putExtra("name",name.text.toString())
-        startActivity(intent)
+    // on click getButton the email address associated with the phone number
+    fun get(view: View) {
+        // get the email address from the mail field
+        val mail = findViewById<TextView>(R.id.mailBox)
+        val phone = findViewById<TextView>(R.id.phoneBox)
 
+        val future = CompletableFuture<String>()
+        val db: DatabaseReference = Firebase.database.reference
 
+        db.child(phone.text.toString()).get().addOnSuccessListener {
+            if (it.value == null) future.completeExceptionally(NoSuchFieldException())
+            else future.complete(it.value as String)
+        }.addOnFailureListener {
+            future.completeExceptionally(it)
+        }
+        future.thenAccept {
+            mail.text = it
+        }
     }
+
+    // set phone and email
+    fun set(view: View) {
```

Comment:
Maybe give to such Function a more descriptive name, at the begening I tough It was something related with only firebase, and i had to see XML files to understand purpose